keyboards/yanghu/unicorne/keymaps/default/oled.c

```diff
@@ -0,0 +1,139 @@
+      0x8a, 0x8b, 0x8c, 0x8d, 10,
+      0xaa, 0xab, 0xac, 0xad, 10,
+      0xca, 0xcb, 0xcc, 0xcd, 10, 0};
+    static const char PROGMEM qmk_logo[] = {
+      0x8e, 0x8f, 0x90, 0x91, 10,
+      0xae, 0xaf, 0xb0, 0xb1, 10,
+      0xce, 0xcf, 0xd0, 0xd1, 10, 0};
+    // clang-format on
+    if (IS_LAYER_ON(_LOWER)) {
+        oled_write_P(symbol_logo, false);
+    } else if (IS_LAYER_ON(_RAISE)) {
+        oled_write_P(numlock_logo, false);
+    } else if (IS_LAYER_ON(_BASE)) {
+        oled_write_P(qmk_logo, false);
+    } else {
+        if (IS_LAYER_ON(_ADJUST)) {
+            oled_write_ln("ADJ", false);
+        } else {
+            oled_write_ln("?????", false);
+        }
+        oled_write_ln(" ", false);
+        oled_write_ln(" ", false);
+        oled_write_ln(" ", false);
+    }
+}
+
+// Oneshot mods status
+uint8_t osmods;
+
+void oled_render_mods(void) {
+    static const char PROGMEM ctrl[]     = {0xb6, 0xb7, 10, 0xd6, 0xd7, 10, 0x20, 0x20, 10, 0};
+    static const char PROGMEM shift[]    = {0x87, 0x88, 0x89, 10, 0xa7, 0xa8, 0xa9, 10, 0xc7, 0xc8, 0xc9, 10, 0};
+    static const char PROGMEM alt[]      = {0x84, 0x85, 0x86, 10, 0xa4, 0xa5, 0xa6, 10, 0xc4, 0xc5, 0xc6, 10, 0};
+    static const char PROGMEM ctrl_alt[] = {0xb6, 0xb7, 0x84, 0x85, 0x86, 0xd6, 0xd7, 0xa4, 0xa5, 0xa6, 0x20, 0x20, 0xc4, 0xc5, 0xc6, 0};
+
+    static const char PROGMEM ctrl_shift[] = {0xb6, 0xb7, 0x87, 0x88, 0x89, 0xd6, 0xd7, 0xa7, 0xa8, 0xa9, 0x20, 0x20, 0xc7, 0xc8, 0xc9, 0};
+    static const char PROGMEM c_a_shift[]  = {0xb6, 0xb7, 0x84, 0x85, 0x86, 0xd6, 0xd7, 0xa4, 0xa5, 0xa6, 0x20, 0x20, 0xc4, 0xc5, 0xc6, 0x87, 0x88, 0x89, 10, 0xa7, 0xa8, 0xa9, 10, 0xc7, 0xc8, 0xc9, 10, 0};
+    // Now check mod status and render.
+    static uint8_t mods;
+    mods = get_mods() | osmods;
+    if ((mods & MOD_MASK_CTRL) && (mods & MOD_MASK_ALT) && (mods & MOD_MASK_SHIFT)) {
+        oled_write_P(c_a_shift, false);
+    } else if ((mods & MOD_MASK_CTRL) && (mods & MOD_MASK_ALT)) {
+        oled_write_P(ctrl_alt, false);
+    } else if ((mods & MOD_MASK_CTRL) && (mods & MOD_MASK_SHIFT)) {
+        oled_write_P(ctrl_shift, false);
+    } else if ((mods & MOD_MASK_SHIFT) && (mods & MOD_MASK_ALT)) {
+        oled_write_P(alt, false);
+        oled_write_P(shift, false);
+    } else if (mods & MOD_MASK_CTRL) {
+        oled_write_P(ctrl, false);
+    } else if (mods & MOD_MASK_ALT) {
+        oled_write_P(alt, false);
+    } else if (mods & MOD_MASK_SHIFT) {
+        oled_write_P(shift, false);
+    } else {
+        for (int i = 0; i < 6; ++i) {
+            oled_write_ln(" ", false);
+        }
+    }
+    return;
+}
+
+void oneshot_mods_changed_user(uint8_t mods) { osmods = mods; }
+
+void oled_task_user(void) {
+    /* oled_render_keylog(); */
+    oled_render_layer();
+    oled_render_mods();
+}
+
+// Call this from "led_update_user" and use `led_state.caps_lock` to check
+// the status of capslock.
+void oled_render_capslock(bool caps_on) {
```

Comment:
This should be done in the oled_task_user function

You can use `host_keyboard_led_state()` to get the current status.

Reply:
fixed in c4ea096